### PR TITLE
(SIMP-4680) Add simp_banners module

### PR DESCRIPTION
**Pull request by trevor-vaughan**
This module has been created to hold common 'Internet Banners' that are
publicly available without restriction.

Different modules need the content of these banners so it was pulled out
into a separate module for ease of use and to allow the repository to
move independely of the requiring modules.

SIMP-4736 #close

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,275 @@
+---
```
Comment (lnemsick-simp) [inline .gitlab-ci.yml]: This doesn't look like the gitlab-ci.yml we are predominantly using

File: .puppet-lint.rc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+--log-format="%{path}:%{line}:%{check}:%{KIND}:%{message}"
+--relative
+--no-class_inherits_from_params_class-check
```
Comment (lnemsick-simp) [inline .puppet-lint.rc]: Do you need this line?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,143 @@
-# pupmod-simp-simp_banners
-A collection of banners that can be used in other Puppet modules
+**FIXME**: Ensure the badges are correct and complete, then remove this message!
```
Comment (lnemsick-simp) [inline README.md]: Looks like this is simply a template.  Please fill in/address FIXMEs.

File: metadata.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  "license": "Apache-2.0",
+  "source": "https://github.com/simp/pupmod-simp-simp_banners",
+  "project_page": "https://github.com/simp/pupmod-simp-simp_banners",
+  "issues_url": "https://simp-project.atlassian.net",
+  "tags": [
+    "simp"
+  ],
+  "dependencies": [
+    {
+      "name": "simp/iptables",
```
Comment (lnemsick-simp) [inline metadata.json]: Why is iptables a dependency?

File: metadata.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      "version_requirement": ">= 6.0.1 < 7.0.0"
+    },
+    {
+      "name": "simp/simplib",
+      "version_requirement": ">= 3.3.1 < 4.0.0"
```
Comment (lnemsick-simp) [inline metadata.json]: Why is simplib dependency?

File: metadata.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    {
+      "name": "puppetlabs/stdlib",
+      "version_requirement": ">= 4.13.1 < 5.0.0"
```
Comment (lnemsick-simp) [inline metadata.json]: How is stdlib used?

File: spec/acceptance/suites/default/class_spec.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    it 'should work with no errors' do
+      apply_manifest(manifest, :catch_failures => true)
+    end
+
+    it 'should be idempotent' do
+      apply_manifest(manifest, :catch_changes => true)
+    end
+
+
+    describe package('simp_banners') do
```
Comment (lnemsick-simp) [inline spec/acceptance/suites/default/class_spec.rb]: ??

File: spec/acceptance/suites/default/class_spec.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+      it { is_expected.to be_installed }
+    end
+
+    describe service('simp_banners') do
```
Comment (lnemsick-simp) [inline spec/acceptance/suites/default/class_spec.rb]: ??

File: spec/acceptance/suites/default/00_default_spec.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
+require 'spec_helper_acceptance'
+
+test_name '<%= metadata.name %> class'
```
Comment (lnemsick-simp) [inline spec/acceptance/suites/default/00_default_spec.rb]: looks like an erb <%= metadata.name *>?

File: .fixtures.yml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+---
+fixtures:
+  repositories:
+    simplib:
+      repo: https://github.com/simp/pupmod-simp-simplib
```
Comment (lnemsick-simp) [inline .fixtures.yml]: Are these fixtures actually needed?

File: lib/puppet/functions/simp_banners/fetch.rb [hunk 11]
```diff
@@ -0,0 +1,77 @@
+# Retrieve a banner from the known list and format it according to the passed
+# options
+#
+Puppet::Functions.create_function(:'simp_banners::fetch') do
```
Comment (lnemsick-simp) [inline lib/puppet/functions/simp_banners/fetch.rb]: simp_banners::select seems more intuitive than simp_banners::fetch?

File: .fixtures.yml (hunk 10 above)
Comment (op-ct) [reply .fixtures.yml]: Not that I can see.

File: spec/acceptance/suites/default/00_default_spec.rb (hunk 9 above)
Comment (op-ct) [reply spec/acceptance/suites/default/00_default_spec.rb]: None of the  skeleton's boilerplate  tests make much sense in a module without any classes.

I've removed them for now.

File: metadata.json (hunk 4 above)
Comment (op-ct) [reply metadata.json]: Removed

File: metadata.json (hunk 5 above)
Comment (op-ct) [reply metadata.json]: Removed

File: metadata.json (hunk 6 above)
Comment (op-ct) [reply metadata.json]: The stdlib function `get_module_path` is used in `fetch.rb`

File: lib/puppet/functions/simp_banners/fetch.rb (hunk 11 above)
Comment (op-ct) [reply lib/puppet/functions/simp_banners/fetch.rb]: I would expect a method named `fetch` more likely to return something than one named `select`. 

(And yet there's `Hash.select`, so :man_shrugging: )

File: .puppet-lint.rc (hunk 2 above)
Comment (op-ct) [reply .puppet-lint.rc]: nope!  (removed)

**Pull request merged**